pkg/handlers/config.go

```diff
@@ -64,6 +65,86 @@ type CurrentAppConfigResponse struct {
 	ConfigGroups []kotsv1beta1.ConfigGroup `json:"configGroups"`
 }
 
+type DownloadAppConfigResponse struct {
+	Success bool   `json:"success"`
+	Error   string `json:"error,omitempty"`
+}
+
+func (h *Handler) DownloadAppConfig(w http.ResponseWriter, r *http.Request) {
+	downloadAppConfigResponse := DownloadAppConfigResponse{
+		Success: false,
+	}
+
+	foundApp, err := store.GetStore().GetAppFromSlug(mux.Vars(r)["appSlug"])
+	if err != nil {
+		logger.Error(err)
+		downloadAppConfigResponse.Error = "failed to get app from app slug"
+		JSON(w, http.StatusInternalServerError, downloadAppConfigResponse)
+		return
+	}
+
+	sequence, err := strconv.Atoi(mux.Vars(r)["sequence"])
+	if err != nil {
+		logger.Error(err)
+		downloadAppConfigResponse.Error = "failed to parse app sequence"
+		JSON(w, http.StatusInternalServerError, downloadAppConfigResponse)
+		return
+	}
+
+	filename := mux.Vars(r)["filename"]
+	if filename == "" {
+		logger.Error(err)
+		downloadAppConfigResponse.Error = "failed to parse filename, parameter was empty"
+		JSON(w, http.StatusInternalServerError, downloadAppConfigResponse)
+		return
+	}
+
+	archiveDir, err := ioutil.TempDir("", "kotsadmconfig")
+	if err != nil {
+		logger.Error(err)
+		downloadAppConfigResponse.Error = "failed to create temp directory"
+		JSON(w, http.StatusInternalServerError, downloadAppConfigResponse)
+	}
+	defer os.RemoveAll(archiveDir)
+
+	configValue, err := getAppConfigValue(foundApp, int64(sequence), filename, archiveDir)
+	if err != nil {
+		logger.Error(err)
+		downloadAppConfigResponse.Error = "failed to get app config"
+		JSON(w, http.StatusInternalServerError, downloadAppConfigResponse)
+		return
+	}
+
+	decoded, err := base64.StdEncoding.DecodeString(configValue)
```

Comment:
also consider using `Decode` instead of `DecodeString` since files can be binary and that would be cleaner and also would match `Content-Type` header better set at the end of this function.

Reply:
addressed